torch/distributed/elastic/utils/data/elastic_distributed_sampler.py

```diff
@@ -52,10 +52,10 @@ def __iter__(self):
         g = torch.Generator()
         g.manual_seed(self.epoch)
         indices = (
-            torch.randperm(len(self.dataset) - self.start_index, generator=g)  # type: ignore[arg-type]
```

Comment:
is `start_index` supposed to mean the starting index from the original dataset on which to sample? Or the starting index for the subset of the iteration. The class comments suggest the former, but you changes are for the latter

Reply:
@H-Huang 
I think the starting index should be applied to the subset of iterations of the current epoch. Only in this way can the correct sample be used when the job is restarted. Maybe we should change the comment for start_index.